### PR TITLE
DDF-4194 add section on contributing documentation to docs

### PR DESCRIPTION
**Pull request by ricklarsen**
#### What does this PR do?

adds a section to developer documentation to describe process for updating documentation.

#### Who is reviewing it? 

@mcalcote @ahoffer @austinsteffes @garrettfreibott 

#### Relevant Component Team: 

@codice/docs 

#### Ask 2 committers to review/merge the PR and tag them here.

@clockard
@lessarderic
@ricklarsen - Documentation
@shaundmorris

#### How should this be tested?

- builds successfully with content included
- review content

#### What are the relevant tickets?
[DDF-4194](https://codice.atlassian.net/browse/DDF-4194)

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: distribution/docs/src/main/resources/content/_devComponents/editing-docs.adoc [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+If creating a new section is required, there are some minimal requirement for a new `.adoc` file.
+
+.Header content
+The templates scan the header information to place it into the correct place within the documentation.
+Different sections have different headers required, but some common attributes are always required.
+
+* `type`: roughly maps to the section or subSection of the documentation.
+* `title`: title of the section or subsection contained in the file.
+* `status`: set to `published` to include within the documentation, set to `draft` to hide a work-in-progess section.
```
[the remark this answers is not in the document]
Comment (ricklarsen) [reply distribution/docs/src/main/resources/content/_devComponents/editing-docs.adoc]: fixed

**Comment by ricklarsen** [comment]
build now

File: distribution/docs/src/main/resources/content/_devComponents/editing-docs.adoc [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+|Template files used to compile the documentation for display.
+
+|jbake.properties
+|Properties file defining content types and other parameters.
+|===
+
+==== Editing Existing Documentation
+
+Update existing content when code behavior changes, new capabilities are added to features, or the configuration process changes.
+Content is organized within the `content` directory in sub directories according the audience and purpose for each section of the documentation.
```
Comment (mdang8) [inline distribution/docs/src/main/resources/content/_devComponents/editing-docs.adoc]: Typo: `sub directories according the audience` should have `to` between `according` and `the`.

File: distribution/docs/src/main/resources/content/_devComponents/editing-docs.adoc [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Reference:: This section is a comprehensive list of features and possible configurations.
+
+Metadata Reference:: This section details how metadata is extracted and normalized by ${branding}.
+
+See the https://codice.atlassian.net/wiki/spaces/DDF/pages/6291516/Documentation+Style+Guide[style guide] for more guidance on stylistic and formatting concerns.
+
+==== Adding New Documentation Content
+
+If creating a new section is required, there are some minimal requirement for a new `.adoc` file.
```
Comment (mdang8) [inline distribution/docs/src/main/resources/content/_devComponents/editing-docs.adoc]: Typo: `requirement` should be `requirements`

**Comment by ricklarsen** [comment]
build now

**Pull request merged**